src/view/view-updater.ts

```diff
@@ -1,27 +1,37 @@
 import { Injectable, Type } from '@nestjs/common';
-import { IViewUpdater } from './interfaces/view-updater';
+import { IViewUpdater } from './interfaces';
 import { IEvent } from '@nestjs/cqrs';
 import { ModuleRef } from '@nestjs/core';
 import { ViewUpdaters } from './view-updaters';
+import { AggregateRootAsync } from '../aggregate-root-async';
 
 @Injectable()
 export class ViewUpdater {
+  private instances = new Map<
+    Type<IViewUpdater<IEvent>>,
+    IViewUpdater<IEvent>
+  >();
 
-    private instances = new Map<Type<IViewUpdater<IEvent>>, IViewUpdater<IEvent>>();
+  constructor(private moduleRef: ModuleRef) {}
 
-    constructor(
-        private moduleRef: ModuleRef,
-    ) {
-    }
-
-    async run<T extends IEvent>(event: T): Promise<void> {
-        const updater = ViewUpdaters.get(event.constructor.name);
-        if (updater) {
-            if (!this.instances.has(updater)) {
-                this.instances.set(updater, this.moduleRef.get(updater.name, { strict: false }));
-            }
-            await this.instances.get(updater).handle(event);
-        }
-        return;
+  async run<T extends IEvent>(
+    event: T,
+    rootAggregator?: AggregateRootAsync,
+  ): Promise<void> {
+    const updater = ViewUpdaters.get(event.constructor.name);
+    if (updater) {
+      if (!this.instances.has(updater)) {
+        this.instances.set(
+          updater,
+          this.moduleRef.get(updater.name, { strict: false }),
+        );
+      }
+      if (rootAggregator !== undefined) {
+        await this.instances.get(updater).handle(event, rootAggregator);
```

Comment:
Handler functions currently implement the `IEventHandler` interface from `@nestjs/cqrs`, which doesn't tell the developer that there's a second argument that can be passed to the handle method.
Maybe, we need to create and export an additional interface that takes this into account.

<img width="676" alt="Screen Shot 2020-11-05 at 4 42 33 PM" src="https://user-images.githubusercontent.com/7079419/98234032-de8fa780-1f85-11eb-936b-f1e0cd96e9df.png">

Reply:
@spunkypanda In this place handle method from IViewUpdater interface, that extended for using `rootAggregator`